libminifi/include/utils/ScopeGuard.h

```diff
@@ -33,10 +33,6 @@ namespace utils {
 
 struct ScopeGuard : ::gsl::final_action<std::function<void()>> {
   using ::gsl::final_action<std::function<void()>>::final_action;
-
-  void disable() noexcept {
-    dismiss();
-  }
```

Comment:
I wouldn't remove this just for the sake of removing it, because external code might depend on it, and it's not a huge maintenance burden to keep it.

Reply:
I do not like this pattern being present in the codebase at all. Restored, but marked for depracation ASAP (1.0).

Reply:
Ok. I didn't think this was harmful, but if you think so, then deprecation sounds like a good idea.